core/src/main/scala/kafka/log/Log.scala

```diff
@@ -181,32 +188,54 @@ class Log(val dir: File,
         // if its a log file, load the corresponding log segment
         val start = filename.substring(0, filename.length - LogFileSuffix.length).toLong
         val indexFile = Log.indexFilename(dir, start)
+        val timeIndexFile = Log.timeIndexFilename(dir, start)
         val segment = new LogSegment(dir = dir,
                                      startOffset = start,
                                      indexIntervalBytes = config.indexInterval,
                                      maxIndexSize = config.maxIndexSize,
                                      rollJitterMs = config.randomSegmentJitter,
                                      time = time,
                                      fileAlreadyExists = true)
-
-        if(indexFile.exists()) {
+        if (indexFile.exists() && timeIndexFile.exists()) {
```

Comment:
That's a good point. It seems that we should check the existence of the index file before creating LogSegment.